code.cpp

```diff
@@ -0,0 +1,136 @@
+#include <array>
+#include <cmath>
+#include <cstdlib>
+#include <iostream>
+#include <unistd.h>
+#include <vector>
+
+const int length = 32;
+const int hight = 11; 
+const int tabl_size = length * hight;
+
+std::array<int,tabl_size> null_array() { 
+    std::array<int,tabl_size> null;
+    for (int i = 0; i < tabl_size; i++) {
+        null[i] = 0;
+    }
+    return null;
+}
+
+struct ListNode {
+    std::array<int,tabl_size> val;
+    ListNode* next;
+    ListNode(): val(null_array()), next(nullptr) {}
+    ListNode(std::array<int,tabl_size> x): val(x), next(nullptr) {}
+    ListNode(std::array<int,tabl_size> x, ListNode* n): val(x), next(n) {}
+};
+
+std::array<int,tabl_size> step(std::array<int,tabl_size> arr) {
+    std::array<int, tabl_size> res; 
+    for (int i = 0; i < hight; i++) {
+        for (int j = 0; j < length; j++) {
+            int count = 0;
+            int k = 0, n = 0;
+            int l = 3, m = 3;
+            if (i == 0) {
```

Comment:
I think that iterating through cell neighbors can be implemented more readable. I think cycles are not necessary here